openvino_xai/methods/black_box/aise.py

```diff
@@ -0,0 +1,221 @@
+    :param device_name: Device type name.
+    :type device_name: str
+    :param prepare_model: Loading (compiling) the model prior to inference.
+    :type prepare_model: bool
+    """
+
+    def __init__(
+        self,
+        model: ov.Model,
+        postprocess_fn: Callable[[OVDict], np.ndarray],
+        preprocess_fn: Callable[[np.ndarray], np.ndarray] = IdentityPreprocessFN(),
+        device_name: str = "CPU",
+        prepare_model: bool = True,
+    ):
+        super().__init__(model=model, preprocess_fn=preprocess_fn, device_name=device_name)
+        self.postprocess_fn = postprocess_fn
+
+        # TODO: resete state ever run
+        self.data_preprocessed = None
+        self.target = None
+        self.num_iterations_per_kernel = None
+        self.kernel_width = None
+        self.seed = None
+
+        self._kernel_params_hist = collections.defaultdict(list)
+        self._pred_score_hist = collections.defaultdict(list)
+        self.input_size = None
+        self._mask_generator = None
+        
+        if prepare_model:
+            self.prepare_model()
+
+    def generate_saliency_map(
+        self,
+        data: np.ndarray,
+        explain_target_indices: List[int] | None = None,
+        preset: Preset = Preset.BALANCE,
+        num_iterations_per_kernel: int | None = None,
+        kernel_widths: List[float] | np.ndarray | None = None,
+        seed: int = 0,
+        scale_output: bool = True,
+    ):
+        """
+        Generates inference result of the RISE algorithm.
+
+        :param data: Input image.
+        :type data: np.ndarray
+        :param explain_target_indices: List of target indices to explain.
+        :type explain_target_indices: List[int]
+        :param num_iterations_per_kernel: TBD.
+        :type num_iterations_per_kernel: int
+        :param seed: TBD.
+        :type seed: int
+        :param scale_output: Whether to scale output or not.
+        :type scale_output: bool
+        """
+        # TODO: support multiple explain_target_indices
+        if len(explain_target_indices) > 1:
+            raise ValueError
+
+        self.data_preprocessed = self.preprocess_fn(data)
+        self.target = explain_target_indices[0]
+        self._preset_parameters(preset, num_iterations_per_kernel, kernel_widths)
+        self.seed = seed
+
+        self.input_size = infer_size_from_image(self.data_preprocessed)
+        self._mask_generator = GaussianPerturbationMask(self.input_size)
+        
+        saliency_maps = self._run_synchronous_explanation()
+
+        if scale_output:
+            saliency_maps = scaling(saliency_maps)
+        return {explain_target_indices[0]: saliency_maps}
+
+    def _preset_parameters(
+            self, 
+            preset: Preset, 
+            num_iterations_per_kernel: int | None, 
+            kernel_widths: List[float] | np.ndarray | None,
+        ) -> None:
+        if preset == Preset.SPEED:
```

Comment:
Great idea with presets! We can do something like that for RISE in future

Reply:
Sure, I will update RISE as well in the next PR